accord-core/src/main/java/accord/primitives/PartialDeps.java

```diff
@@ -0,0 +1,87 @@
+package accord.primitives;
+
+public class PartialDeps extends Deps
+{
+    public static final PartialDeps NONE = new PartialDeps(KeyRanges.EMPTY, Deps.NONE.keys, Deps.NONE.txnIds, Deps.NONE.keyToTxnId);
+
+    public static class SerializerSupport
+    {
+        private SerializerSupport() {}
+
+        public static PartialDeps create(KeyRanges covering, Keys keys, TxnId[] txnIds, int[] keyToTxnId)
+        {
+            return new PartialDeps(covering, keys, txnIds, keyToTxnId);
+        }
+    }
+
+    public static class OrderedBuilder extends AbstractOrderedBuilder<PartialDeps>
+    {
+        final KeyRanges covering;
+        public OrderedBuilder(KeyRanges covering, boolean hasOrderedTxnId)
+        {
+            super(hasOrderedTxnId);
+            this.covering = covering;
+        }
+
+        @Override
+        PartialDeps build(Keys keys, TxnId[] txnIds, int[] keysToTxnIds)
+        {
+            return new PartialDeps(covering, keys, txnIds, keysToTxnIds);
+        }
+    }
+
+    public static OrderedBuilder orderedBuilder(KeyRanges ranges, boolean hasOrderedTxnId)
+    {
+        return new OrderedBuilder(ranges, hasOrderedTxnId);
+    }
+
+    public final KeyRanges covering;
+
+    PartialDeps(KeyRanges covering, Keys keys, TxnId[] txnIds, int[] keyToTxnId)
+    {
+        super(keys, txnIds, keyToTxnId);
+        this.covering = covering;
```

Comment:
Should this assert covering actually covers everything?

Reply:
covering is intended to serve as a minimum, not a maximum, though we could perhaps treat it as a precise property.

Reply:
If there is a performance argument not to do it then sure don't do it. Otherwise it feels like it should be precise to make sure we aren't shipping extra data due to a bug?

Reply:
Sure, ok.